### DISCUSSION TITLE
Set/Send the SOC by MQTT

### DISCUSSION THREAD

**Discussion by teicors** [celsworth/lxp-bridge#276] [General]
Hi,
I'm very sad in discovering the great Chris's site in "read only mode". Great work Chris ! 

I have an HW converter to inform my inverter the battery's SOC, but I'm wondering if is possible to send an MQTT message to send the info ?
Cheers,
Angelo

**Comment by celsworth** [comment]
Hey!

There hasn't been much work done on lxp-bridge for a while and I've mostly moved onto new projects, also I don't have a Luxpower inverter anymore so I wouldn't be able to properly test any new changes.

To answer your question, not currently - the SOC is sent to Luxpower inverters from managed batteries (like Pylontechs) over CANBUS; the Wifi dongle as far as I'm aware has no access to that so we can't send CANBUS values to the inverter using the existing TCP connection we have to it. The SOC does end up in a read-only input register so we can see it, but we can't set it.

To support this we'd need to pretend to be a managed battery and open a CANBUS port (usually some extra piece of hardware) which is then physically connected to the inverter. We could then send all the data that a Pylontech usually sends to the inverter; basically emulating being a battery. It's an interesting idea but would probably be quite a lot of work and I'm not sure how much interest there would be as the use-case would be limited I think.
